### PR TITLE
MINOR: Remind user index file is empty when dumping LogSegment index file

### PR DESCRIPTION
**Pull request by 17hao**
Remind user index file is empty when dumping LogSegment xxx.index file instead of throwing an `NoSuchElementException`.

BEFORE
image

AFTER
image

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by chia7712 — COMMENTED** [review]
@17hao Thanks for your patch. Is it possible to offer a unit test? Thanks!

**Comment by 17hao** [comment]
hi @chia7712 ,  I added an ut, PTAL.

File: core/src/test/scala/unit/kafka/tools/DumpLogSegmentsTest.scala [hunk 1]
```diff
@@ -168,6 +168,19 @@ class DumpLogSegmentsTest {
...
+  def testDumpEmptyIndex(): Unit = {
+    val indexFile = new File(indexFilePath)
+    new PrintWriter(indexFile).close()
+    val expectOutput = s"$indexFile is empty.\n"
+    val outContent = new ByteArrayOutputStream()
+    Console.withOut(outContent) {
+      DumpLogSegments.dumpIndex(indexFile, indexSanityOnly = false, verifyOnly = true,
+        misMatchesForIndexFilesMap = mutable.Map[String, List[(Long, Long)]](), Int.MaxValue)
+    }
+    assertEquals(outContent.toString, expectOutput)
```
Comment (chia7712) [inline core/src/test/scala/unit/kafka/tools/DumpLogSegmentsTest.scala]: ```expectOutput``` should be moved to first argument.

**Review by chia7712 — CHANGES_REQUESTED** [review]
@17hao Thanks for your patch. Please fix a minor comment.

File: core/src/test/scala/unit/kafka/tools/DumpLogSegmentsTest.scala (hunk 1 above)
Comment (17hao) [reply core/src/test/scala/unit/kafka/tools/DumpLogSegmentsTest.scala]: My bad, I fixed it. Thanks!

**Review by chia7712 — APPROVED** [review]
@17hao Thanks for your patch. LGTM

**Comment by chia7712** [comment]
@17hao Could you merge trunk to include recent fixes of flaky tests?

**Comment by 17hao** [comment]
@chia7712  Ok, I merged trunk :)

**Comment by chia7712** [comment]
@17hao Could you merge trunk to include recent hot fix? Sorry that we don't keep our tests stable :(

**Comment by 17hao** [comment]
@chia7712 Done😂

**Comment by chia7712** [comment]
@17hao Thanks for your patch!

**Pull request merged**